Geometry/HGCalMapping/plugins/alpaka/HGCalMappingCellESProducer.cc

```diff
@@ -0,0 +1,153 @@
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/host.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "CondFormats/DataRecord/interface/HGCalMappingCellIndexerRcd.h"
+#include "CondFormats/DataRecord/interface/HGCalMappingCellRcd.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingCellIndexer.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingParameterHostCollection.h"
+#include "CondFormats/HGCalObjects/interface/alpaka/HGCalMappingParameterDeviceCollection.h"
+#include "DataFormats/HGCalDigi/interface/HGCalElectronicsId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCScintillatorDetId.h"
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  namespace hgcal {
+
+    class HGCalMappingCellESProducer : public ESProducer {
+    public:
+      //
+      HGCalMappingCellESProducer(const edm::ParameterSet& iConfig)
+          : ESProducer(iConfig), filelist_(iConfig.getParameter<std::vector<std::string> >("filelist")) {
+        auto cc = setWhatProduced(this);
+        cellIndexTkn_ = cc.consumes(iConfig.getParameter<edm::ESInputTag>("cellindexer"));
+      }
+
+      //
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+        edm::ParameterSetDescription desc;
+        desc.add<std::vector<std::string> >("filelist", std::vector<std::string>({}))
+            ->setComment("list of files with the readout cells of each module");
+        desc.add<edm::ESInputTag>("cellindexer", edm::ESInputTag(""))->setComment("Dense cell index tool");
+        descriptions.addWithDefaultLabel(desc);
+      }
+
+      //
+      std::optional<HGCalMappingCellParamHostCollection> produce(const HGCalMappingCellRcd& iRecord) {
+        //get cell indexer
+        const HGCalMappingCellIndexer& cellIndexer = iRecord.get(cellIndexTkn_);
+
+        const uint32_t size = cellIndexer.maxDenseIndex();  // channel-level size
+        HGCalMappingCellParamHostCollection cellParams(size, cms::alpakatools::host());
+        for (uint32_t i = 0; i < size; i++)
+          cellParams.view()[i].valid() = false;
+
+        for (const auto& filename : filelist_) {
+          //open file and read the first line to identify which type it is
+          edm::FileInPath fip(filename);
+          std::ifstream file(fip.fullPath());
```

Comment:
(this question concerns also `HGCalMappingModuleESProducer`)

What is the eventual plan for the storage of this data currently read from files? Would it be stored in CondDB? In the present system that would imply another data product + Record pair (assuming it would not make sense to store the data in `HGCalMappingCellIndexer`), and the `HGCalMappingCellRcd` would be made to depend on that Record.

(if the plan would be along these lines, that would counter my earlier guess in https://github.com/cms-sw/cmssw/pull/43751/files#r1497864199 about the fate of `HGCalMappingCellRcd` / `HGCalMappingModuleRcd`)

Reply:
Once we have all the hexa/tileboards designed/mapped (sometime this year) the two files will be final. It means that HGCalMappingCellRcd and HGCalMappingCellIndexerRcd can be filled and stored as they are and we don't need to touch them again*

For the other two HGCalMapping{Module,ModuleIndexer}Rcd records, the situation is trickier and i can't forecast yet what's going to happen. We'll probably need files for a long time to maintain:
- a viable mapping scenario for simulation purposes
- ability to inject different mappings to accompany the plethora of system tests foreseen before installation in P5
 

* unless in the long run we find that the current approach is sub-optimal or some additional info is needed, but that we'll only know by exercising the system